src/components/BookList/BookList.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { BOOK_LIST } from '../../pages/ProductDetail/ProductDetailData';
+import * as s from '../../pages/ProductDetail/ProductDetailStyled';
+
+const BookList = () => {
+  const [bookListData, setBookListData] = useState([]);
+  const [checkedList, setCheckedList] = useState([]);
+
+  const navigate = useNavigate();
+
+  const handleSingleCheck = (checked, id) => {
+    if (checked) {
+      setCheckedList(prev => [...prev, id]);
+    } else {
+      setCheckedList(checkedList.filter(el => el !== id));
+    }
+  };
+
+  const handleAllCheck = checked => {
+    if (checked) {
+      const idArr = [];
+      BOOK_LIST.forEach(el => idArr.push(el.id));
+      setCheckedList(idArr);
+    } else {
+      setCheckedList([]);
+    }
+  };
+
+  useEffect(() => {
+    fetch('API address')
+      .then(response => response.json())
+      .then(data => setBookListData(data));
+  }, ['params id']);
+
+  const clickToOrder = e => {
+    fetch('API address', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        Authorization: localStorage.getItem('TOKEN'),
+      },
+      body: {
+        // checkedList의 책id, 각 권의 id
+      },
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.length === 0) {
+          alert('책을 선택해주세요.');
+        }
+      });
+
+    navigate('/bookcase');
+  };
+
+  return (
+    <s.DetailBodySeries>
+      <s.SeriesSelectAll>
+        <div>
+          <input
+            type="checkbox"
+            id="selectAll"
+            onChange={e => {
+              handleAllCheck(e.target.checked);
+            }}
+            checked={checkedList.length === BOOK_LIST.length ? true : false}
```

Comment:
```suggestion
            checked={checkedList.length === BOOK_LIST.length}
```
조건의 반환값은 결국 boolean 값을 내보내주기 때문에 이와 같이 하셔도 되겠습니다!